src/mod_muc.erl

```diff
@@ -733,14 +705,14 @@ load_permanent_rooms(Host, ServerHost, Access, HistorySize, RoomShaper, HttpAuth
 start_new_room(Host, ServerHost, Access, Room,
                HistorySize, RoomShaper, HttpAuthPool, From,
                Nick, DefRoomOpts) ->
-    case mnesia:dirty_read(muc_room, {Room, Host}) of
-        [] ->
+    case mod_muc_db_backend:restore_room(ServerHost, Host, Room) of
+        {error, _} ->
```

Comment:
This is different behaviour than original. Originally, we'd create a room if it was not found; here we create it on all possible errors.